corehq/apps/userreports/sql/connection.py

```diff
@@ -1,10 +1,17 @@
+import random
 
+from corehq.sql_db.connections import connection_manager
 
-def get_engine_id(an_object):
+READ_REPLICA_ROLLOUT_FACTOR = 0.01  # 1%
```

Comment:
Not sure what happened to your comment "It would be nice to have this as a localsetting or setting in HQ UI. that way we wouldn't need a deploy to change this and we could have different values for other envs"

100% agree with this but didn't have the time to implement the UI. I'll move it to settings for now.